### PR TITLE
User handling

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/81565 and https://central.tri.be/issues/89966

This ticket puts in place the user handling functionalities required by the ticket.
In so doing there are some notable changes:
* I've restored all the tests to function, they all pass
* Due to PayPal unscrutable IPN to PDT behavior I've put in place code to track and save "lost" transactions; now we're saving transactions in the options table but we can (should) change that implementation to post types; the tests and the code depend upon the abstraction, not the implementation so we are free to do that later
* I've added an `Invalid_PDT` handler to handle the case above
* Since we do not have a "checkout page", PayPal being the checkout page, I've added the code and links to login and/or register in the ticket view; that's the last place we control
* I've added a `custom` argument to TPP transactions to avoid TPP from trying to manage *any* PayPal transaction; it was killing WC and EDD PP transactions too

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | A closing tag is not permitted at the end of a PHP file | [login-before-purchase.php:31](https://github.com/moderntribe/event-tickets/blob/4c1deb0cf5240200bedb9b195a788f2d67906b34/src/views/login-before-purchase.php#L31)

> via jenkins/codesniffer-phpcs

**Comment by lucatume** [comment]
Still a work in progress: hold your review horses

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space before the last parenthesis | [Gateway.php:158](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Gateway.php#L158)
:x: | There must be a single space before the last parenthesis | [Gateway.php:158](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Gateway.php#L158)
:x: | No space before closing parenthesis is prohibited | [Gateway.php:158](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Gateway.php#L158)
:x: | You must have a space after a closing parenthesis | [Gateway.php:158](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Gateway.php#L158)
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)
:x: | There must be a single space after the first parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space before the last parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space before the last parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | Missing space after array opener. | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space after the first parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space after the first parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space before the last parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space after the first parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space before the last parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | Expected 1 space between comma and &quot;'hook'&quot;; 0 found | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | Missing space before array closer. | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | A closing tag is not permitted at the end of a PHP file | [login-before-purchase.php:36](https://github.com/moderntribe/event-tickets/blob/e0e1193322cd003a04dd3f15fa03823c288e1224/src/views/login-before-purchase.php#L36)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)
:x: | There must be a single space after the first parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space before the last parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space before the last parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | Missing space after array opener. | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space after the first parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space after the first parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space before the last parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space after the first parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | There must be a single space before the last parenthesis | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | Expected 1 space between comma and &quot;'hook'&quot;; 0 found | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | Missing space before array closer. | [Main.php:221](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/Tribe/Commerce/PayPal/Main.php#L221)
:x: | A closing tag is not permitted at the end of a PHP file | [login-before-purchase.php:36](https://github.com/moderntribe/event-tickets/blob/21f57b1f4b665c0a0ad641661e2669ed898cb3d4/src/views/login-before-purchase.php#L36)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/6daf09730d2c4085e6d008adaacd35d7a1dd7ef7/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)
:x: | A closing tag is not permitted at the end of a PHP file | [login-before-purchase.php:36](https://github.com/moderntribe/event-tickets/blob/6daf09730d2c4085e6d008adaacd35d7a1dd7ef7/src/views/login-before-purchase.php#L36)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/da45a9d9d34d8825fa26f5a703b1f0b131167983/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/3742433bad766ac5356e1cc693f984c577291f43/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)
:warning: | The method parameter $ticket_id is never used | [Tickets_View.php:136](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/Tribe/Commerce/PayPal/Tickets_View.php#L136)
:warning: | The method parameter $post_id is never used | [Tickets_View.php:136](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/Tribe/Commerce/PayPal/Tickets_View.php#L136)
:x: | There must be a single space after the first parenthesis | [orders-pp-tickets.php:17](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders-pp-tickets.php#L17)
:x: | There must be a single space before the last parenthesis | [orders-pp-tickets.php:17](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders-pp-tickets.php#L17)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_type' | [orders-pp-tickets.php:31](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders-pp-tickets.php#L31)
:x: | You must have a space after a closing parenthesis | [orders-pp-tickets.php:32](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders-pp-tickets.php#L32)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [orders-pp-tickets.php:47](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders-pp-tickets.php#L47)
:x: | You must have a space after a closing parenthesis | [orders-pp-tickets.php:62](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders-pp-tickets.php#L62)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$view' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$i' | [orders-pp-tickets.php:65](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders-pp-tickets.php#L65)
:x: | You must have a space after a closing parenthesis | [orders.php:57](https://github.com/moderntribe/event-tickets/blob/402cb063f966726ff63a33cedcac8d8a54b0f870/src/views/tickets/orders.php#L57)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)
:x: | There must be a single space after the first parenthesis | [orders-pp-tickets.php:17](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders-pp-tickets.php#L17)
:x: | There must be a single space before the last parenthesis | [orders-pp-tickets.php:17](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders-pp-tickets.php#L17)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_type' | [orders-pp-tickets.php:31](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders-pp-tickets.php#L31)
:x: | You must have a space after a closing parenthesis | [orders-pp-tickets.php:32](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders-pp-tickets.php#L32)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [orders-pp-tickets.php:47](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders-pp-tickets.php#L47)
:x: | You must have a space after a closing parenthesis | [orders-pp-tickets.php:62](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders-pp-tickets.php#L62)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$view' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$i' | [orders-pp-tickets.php:65](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders-pp-tickets.php#L65)
:x: | You must have a space after a closing parenthesis | [orders.php:61](https://github.com/moderntribe/event-tickets/blob/bf4b109271669b5851483411e29439aeea7d1e9b/src/views/tickets/orders.php#L61)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)
:x: | There must be a single space after the first parenthesis | [orders-pp-tickets.php:17](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders-pp-tickets.php#L17)
:x: | There must be a single space before the last parenthesis | [orders-pp-tickets.php:17](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders-pp-tickets.php#L17)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_type' | [orders-pp-tickets.php:31](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders-pp-tickets.php#L31)
:x: | You must have a space after a closing parenthesis | [orders-pp-tickets.php:32](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders-pp-tickets.php#L32)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [orders-pp-tickets.php:47](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders-pp-tickets.php#L47)
:x: | You must have a space after a closing parenthesis | [orders-pp-tickets.php:62](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders-pp-tickets.php#L62)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$view' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$i' | [orders-pp-tickets.php:65](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders-pp-tickets.php#L65)
:x: | You must have a space after a closing parenthesis | [orders.php:61](https://github.com/moderntribe/event-tickets/blob/998cd51c62c5f668ea29086c6cdb19d57988560f/src/views/tickets/orders.php#L61)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/cd8dd56df2fa4324b3a6d3cbe0140920ef9f1a78/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_type' | [orders-pp-tickets.php:31](https://github.com/moderntribe/event-tickets/blob/cd8dd56df2fa4324b3a6d3cbe0140920ef9f1a78/src/views/tickets/orders-pp-tickets.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [orders-pp-tickets.php:47](https://github.com/moderntribe/event-tickets/blob/cd8dd56df2fa4324b3a6d3cbe0140920ef9f1a78/src/views/tickets/orders-pp-tickets.php#L47)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$view' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/cd8dd56df2fa4324b3a6d3cbe0140920ef9f1a78/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/cd8dd56df2fa4324b3a6d3cbe0140920ef9f1a78/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/cd8dd56df2fa4324b3a6d3cbe0140920ef9f1a78/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$i' | [orders-pp-tickets.php:65](https://github.com/moderntribe/event-tickets/blob/cd8dd56df2fa4324b3a6d3cbe0140920ef9f1a78/src/views/tickets/orders-pp-tickets.php#L65)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/0b39e7c29fea0185369be7baf016be0e4fe11ea0/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_type' | [orders-pp-tickets.php:31](https://github.com/moderntribe/event-tickets/blob/0b39e7c29fea0185369be7baf016be0e4fe11ea0/src/views/tickets/orders-pp-tickets.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [orders-pp-tickets.php:47](https://github.com/moderntribe/event-tickets/blob/0b39e7c29fea0185369be7baf016be0e4fe11ea0/src/views/tickets/orders-pp-tickets.php#L47)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$view' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/0b39e7c29fea0185369be7baf016be0e4fe11ea0/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/0b39e7c29fea0185369be7baf016be0e4fe11ea0/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/0b39e7c29fea0185369be7baf016be0e4fe11ea0/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$i' | [orders-pp-tickets.php:65](https://github.com/moderntribe/event-tickets/blob/0b39e7c29fea0185369be7baf016be0e4fe11ea0/src/views/tickets/orders-pp-tickets.php#L65)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Commerce/PayPal/Notices.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+class Tribe__Tickets__Commerce__PayPal__Notices {
+
+	public function show_missing_identity_token_notice() {
```
Comment (borkweb) [inline src/Tribe/Commerce/PayPal/Notices.php]: Can you add docblocks to these methods?

File: src/Tribe/Commerce/PayPal/Transaction.php [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	 * @param string $transaction_id
+	 */
+	public function __construct( $transaction_id ) {
+		$this->transaction_id = $transaction_id;
+		$this->status         = self::$undefined_status;
+	}
+
+	/**
+	 * Builds a new instance from data stored in the database or from default data.
+	 *
```
Comment (borkweb) [inline src/Tribe/Commerce/PayPal/Transaction.php]: Add `@since`

File: src/views/tickets/tpp.php [hunk 3]
```diff
@@ -118,14 +119,19 @@ class="tribe-ticket-quantity qty"
 			do_action( 'event_tickets_tpp_after_ticket_row', tribe_events_get_ticket_event( $ticket->id ), $ticket );
 
 		}
+
+		$is_therea_any_message_to_show = ! is_user_logged_in() && ( $must_login || $can_login );
```
Comment (borkweb) [inline src/views/tickets/tpp.php]: Typo in variable name? `$is_there_any_message_to_show`

File: src/views/tickets/tpp.php [hunk 4]
```diff
@@ -118,14 +119,19 @@ class="tribe-ticket-quantity qty"
...
 		?>
 
-		<?php if ( $is_there_any_product_to_sell ) : ?>
+		<?php if ( $is_there_any_product_to_sell && $is_therea_any_message_to_show ) : ?>
```
Comment (borkweb) [inline src/views/tickets/tpp.php]: Typo in variable name? `$is_there_any_message_to_show`

File: src/Tribe/Commerce/PayPal/Handler/IPN.php [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-class Tribe__Tickets__Commerce__PayPal__Handler__IPN {
+class Tribe__Tickets__Commerce__PayPal__Handler__IPN implements Tribe__Tickets__Commerce__PayPal__Handler__Interface{
```
Comment (borkweb) [inline src/Tribe/Commerce/PayPal/Handler/IPN.php]: Add a space between `Interface` and `{`

**Comment by lucatume** [comment]
Ok, good for another pass.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $transaction is never used | [IPN.php:55](https://github.com/moderntribe/event-tickets/blob/2505f054b4909f33d6d47c18d4c32c4a3ee694ed/src/Tribe/Commerce/PayPal/Handler/IPN.php#L55)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_type' | [orders-pp-tickets.php:31](https://github.com/moderntribe/event-tickets/blob/2505f054b4909f33d6d47c18d4c32c4a3ee694ed/src/views/tickets/orders-pp-tickets.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [orders-pp-tickets.php:47](https://github.com/moderntribe/event-tickets/blob/2505f054b4909f33d6d47c18d4c32c4a3ee694ed/src/views/tickets/orders-pp-tickets.php#L47)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$view' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/2505f054b4909f33d6d47c18d4c32c4a3ee694ed/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/2505f054b4909f33d6d47c18d4c32c4a3ee694ed/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$attendee' | [orders-pp-tickets.php:64](https://github.com/moderntribe/event-tickets/blob/2505f054b4909f33d6d47c18d4c32c4a3ee694ed/src/views/tickets/orders-pp-tickets.php#L64)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$i' | [orders-pp-tickets.php:65](https://github.com/moderntribe/event-tickets/blob/2505f054b4909f33d6d47c18d4c32c4a3ee694ed/src/views/tickets/orders-pp-tickets.php#L65)

> via jenkins/codesniffer-phpcs

**Pull request merged**